### PR TITLE
absorb new meroxa-go interface; small doc changes

### PR DESCRIPTION
**Pull request by janelletavares**
# Description of change

absorb new meroxa-go interface; small doc changes

Fixes <GitHub Issue>

# Type of change

- [ ]  New feature
- [ ]  Bug fix
- [ ]  Refactor
- [ ]  Documentation

# How was this tested?

- [ ]  Unit Tests
- [ ]  Tested in staging

# Demo

**Before this pull-request**

_Include how it looked before_

**After this pull-request**

_Include how it looks now_

# Additional references

*Any additional links (if appropriate)*

# Documentation updated

*Make sure that our [documentation](https://docs.meroxa.com/) is accordingly updated when necessary.*

You can do that by opening a pull-request to our (🔒 private, for now) repository: https://github.com/meroxa/meroxa-docs.

✨ In the future, there will be a GitHub action taking care of these updates automatically. ✨

*Provide PR link:*

### CODE REVIEW

File: cmd/meroxa/root/connectors/describe.go
Comment (dianadoherty) [inline cmd/meroxa/root/connectors/describe.go:70]: Not sure if this is part of the refactor, but you might wanna modify the Execute fn: https://github.com/meroxa/cli/pull/217/files#diff-36603a939a11613ba22e3770d59fc2369f352994353cd5838259f60272fe772dR61-R69

```
func (d *Describe) Execute(ctx context.Context) error {
	connector, err := d.client.GetConnectorByNameOrID(ctx, d.args.NameOrID)
	if err != nil {
		return err
	}
...
```

File: cmd/meroxa/root/connectors/logs.go
Comment (dianadoherty) [inline cmd/meroxa/root/connectors/logs.go:83]: In execute, `GetConnectorLogs` should now accept nameOrID https://github.com/meroxa/meroxa-go/pull/78/files#diff-c10a5335e8fa840c34013ae4363be88f6b8c7ee5573945ced446d6a5f4afe5b5R11

File: cmd/meroxa/root/connectors/remove.go
Comment (dianadoherty) [inline cmd/meroxa/root/connectors/remove.go:66]:
```suggestion
	err = r.client.DeleteConnector(ctx, r.args.NameOrID)
```

File: cmd/meroxa/root/connectors/remove.go
Comment (dianadoherty) [inline cmd/meroxa/root/connectors/remove.go:64]: You also don't need to retrieve the connector name above anymore https://github.com/meroxa/cli/pull/217/files#diff-51d616b524f4dc5671edd4e71797ce6a7cf9c1c23775b1d85c069ab03fd29efcR61

**Review by dianadoherty — COMMENTED** [review]
In a couple of places I see reference to NameOrID as just name, it might be worth it to clarify that it can be either

File: cmd/meroxa/root/connectors/describe.go
Comment (janelletavares) [reply cmd/meroxa/root/connectors/describe.go:70]: the Get functions for all entity types are still divided into just id or just name. until we can change the pattern everywhere to one Get function that processes either name or ID, I just left that pattern as is.

File: cmd/meroxa/root/connectors/remove.go
Comment (janelletavares) [reply cmd/meroxa/root/connectors/remove.go:64]: this link isn't working for me

Comment (janelletavares) [reply cmd/meroxa/root/connectors/remove.go:64]: there were a couple of remove functions that return the whole object... i didn't change that pattern. i don't know if anyone has come to rely on that pattern by now.

**Comment by raulb** [comment]
@janelletavares started reviewing this pull-request, but I think it lacks some context on the motivation behind it. Maybe adding a link to the change made in meroxa-go would be useful. I appreciate we're moving away from having the CLI to define their own mocks, but I noticed some other UX changes (messaging, etc) that would be great to detail.

Let me know if I can help with this.

**Comment by janelletavares** [comment]
> @janelletavares started reviewing this pull-request, but I think it lacks some context on the motivation behind it. Maybe adding a link to the change made in meroxa-go would be useful. I appreciate we're moving away from having the CLI to define their own mocks, but I noticed some other UX changes (messaging, etc) that would be great to detail.
> 
> Let me know if I can help with this.

Here are the related PRs, if that helps.
https://github.com/meroxa/meroxa-go/pull/77
https://github.com/meroxa/meroxa-go/pull/78
https://github.com/meroxa/meroxa-go/pull/79

File: cmd/meroxa/global/client.go
Comment (raulb) [inline cmd/meroxa/global/client.go:121]: if `meroxa.Client` is not longer a pointer, how come we could reuse it if we haven't returned it?

File: cmd/meroxa/root/connectors/connect_test.go
Comment (raulb) [inline cmd/meroxa/root/connectors/connect_test.go:88]: Why making this change from `Metadata` to `Type`?

File: cmd/meroxa/root/connectors/connect_test.go
Comment (raulb) [inline cmd/meroxa/root/connectors/connect_test.go:112]: How come we're no longer comparing on `Metadata`?

File: cmd/meroxa/root/connectors/connect_test.go
Comment (raulb) [inline cmd/meroxa/root/connectors/connect_test.go:186]: I don't think this is right. `meroxa connect ` should still require either `--from` or `--to`.

File: cmd/meroxa/root/connectors/remove.go
Comment (raulb) [inline cmd/meroxa/root/connectors/remove.go:61]: we no longer need to make this API request if we can delete passing the name.

File: cmd/meroxa/root/connectors/remove_test.go
Comment (raulb) [inline cmd/meroxa/root/connectors/remove_test.go:76]: This will go away once we remove the API request we no longer need to make.

File: cmd/meroxa/root/pipelines/create.go
Comment (raulb) [inline cmd/meroxa/root/pipelines/create.go:52]: How come we're removing this?

File: cmd/meroxa/root/pipelines/update.go
Comment (raulb) [inline cmd/meroxa/root/pipelines/update.go:49]: Same as `pipelines create`, how come we're removing this?

**Review by raulb — CHANGES_REQUESTED** [review]
Thank you @janelletavares for providing those links with changes made in `meroxa-go`. I still have some questions to what seem to be changing publicly documented flags (`metadata`, and some expected behavior when creating connectors via `meroxa connect`).

Let's pair if you think I'm still missing some context.

File: cmd/meroxa/root/connectors/connect_test.go
Comment (janelletavares) [reply cmd/meroxa/root/connectors/connect_test.go:112]: The value we put in isn't the totality of the value we get out, so it got messy. there may be a way to adjust for that, i didn't spend a lot of time on it.

File: cmd/meroxa/root/connectors/connect_test.go
Comment (janelletavares) [reply cmd/meroxa/root/connectors/connect_test.go]: oh, maybe this is because of moving the pipeline flag code up 20 lines in the next block

File: cmd/meroxa/root/connectors/create.go
Comment (janelletavares) [reply cmd/meroxa/root/connectors/create.go]: Checking the Pipeline flag was about 20 lines below, and I just moved it up. I don't remember why I moved it, maybe it doesn't need to move at all. I don't know why it's checked explicitly and not handled by the required notation. Since input is no longer directly stored in the config map and only assigned to the type field on new line 133, I removed the code that checks if it's non-empty and assigns it to the map.

File: cmd/meroxa/root/connectors/remove.go
Comment (janelletavares) [reply cmd/meroxa/root/connectors/remove.go:61]: we return the whole connector on line 73, that's the only reason that i kept it. it doesn't appear to be documented behavior in the official developer docs.  do you think anyone is relying on that object to be returned? any warnings before we change that behavior?

File: cmd/meroxa/root/pipelines/create.go
Comment (janelletavares) [reply cmd/meroxa/root/pipelines/create.go:52]: When discussing with Diana, she mentioned it was either unused in the pipeline code or had no use case for being exposed to customers.

File: cmd/meroxa/root/pipelines/update.go
Comment (janelletavares) [reply cmd/meroxa/root/pipelines/update.go:49]: I forget which, it's either unused or has no use case for being exposed to customers. The only metadata example I can find in our official developer documentation is around creating a postgres resource.

**Review by janelletavares — COMMENTED** [review]
thanks for the review!

**Comment by raulb** [comment]
@janelletavares Working on the CLI today I realized I'm blocked on some changes made in `meroxa-go`, so I'll need make some of these changes myself to unblock environments' work, unless this pull-request is broken out in different parts. My biggest concern on this pull-request are the user breaking changes that would need to be communicated to customers (removing support on feature flags).

If we could have only one pull-request that could make the CLI work with the latest of meroxa-go would be great, thank you.

File: cmd/meroxa/global/client.go
Comment (raulb) [reply cmd/meroxa/global/client.go:121]: We just talked about this one in a call, so please ignore me.

File: cmd/meroxa/root/connectors/remove.go
Comment (raulb) [reply cmd/meroxa/root/connectors/remove.go:61]: Actually, never mind. Since `DeleteConnector` doesn't return the connector, we'll still need to fetch it for the json format.

File: cmd/meroxa/root/connectors/remove_test.go
Comment (raulb) [reply cmd/meroxa/root/connectors/remove_test.go:76]: See previous message. This is still needed.

**Comment by janelletavares** [comment]
@raulb if the meroxa-go PR and this PR seem good to you, feel free to squash merge, update the go.mod to unblock you.

**Review by raulb — APPROVED** [review]
This LGTM now. We'll evaluate removing the `metadata` flag in a separate pull-request following the usual process:

1. Check usage of this flag to determine a deprecation path (or simply removing it if it hasn't been used)
2. Depending on 1, hide flag, set a deprecation date to remove it altogether
3. Remove support in [meroxa-go](https://github.com/meroxa/meroxa-go)

**Pull request merged**
